homeassistant/components/plant.py

```diff
@@ -97,8 +103,16 @@ def async_setup(hass, config):
         async_track_state_change(hass, sensor_entity_ids, entity.state_changed)
         entities.append(entity)
 
-    yield from component.async_add_entities(entities)
+        if CONF_GROUP_NAME in plant_config and \
+                plant_config[CONF_GROUP_NAME] is not None:
+            group_name = plant_config[CONF_GROUP_NAME]
+            members = ['{}.{}'.format(DOMAIN, plant_name)]  # own entity id
```

Comment:
Instead to generate IDs that you not can sure that the core will use the same IDs, you should only "create" a empty group without entities and give the group name into Entiy object. You can use the `async_added_to_hass` callback to join the entity in the group after the core as written the real entity id to the object.

Reply:
Fixed. 

This part is a bit frustrating for me: This was the 4th time, I've implemented the way groups are created. I hope this now meets of your expectations.